--- origsrc/sugar-toolkit-gtk3-0.106.1/configure.ac	2015-08-26 09:38:33.000000000 -0500
+++ src/sugar-toolkit-gtk3-0.106.1/configure.ac	2015-12-29 23:45:03.116428200 -0600
@@ -18,7 +18,7 @@ AC_PATH_PROG([GLIB_GENMARSHAL], [glib-ge
 PYTHON=python2
 AM_PATH_PYTHON
 
-PKG_CHECK_MODULES(EXT, gtk+-3.0 gdk-3.0 gdk-pixbuf-2.0 sm ice alsa
+PKG_CHECK_MODULES(EXT, gtk+-3.0 gdk-3.0 gdk-pixbuf-2.0 sm ice
                        librsvg-2.0 xfixes xi x11)
 
 GLIB_MKENUMS=`$PKG_CONFIG glib-2.0 --variable=glib_mkenums`
--- origsrc/sugar-toolkit-gtk3-0.106.1/src/sugar3/Makefile.am	2015-01-14 13:13:03.000000000 -0600
+++ src/sugar-toolkit-gtk3-0.106.1/src/sugar3/Makefile.am	2015-12-30 00:17:42.333489000 -0600
@@ -24,7 +24,6 @@ nodist_sugar_PYTHON = config.py
 lib_LTLIBRARIES = libsugarext.la
 
 libsugarext_la_CFLAGS = \
-	-DHAVE_ALSA \
 	$(EXT_CFLAGS) \
 	$(WARN_CFLAGS)
 
@@ -33,8 +32,8 @@ libsugarext_la_CFLAGS = \
 
 libsugarext_la_SOURCES =		\
 	$(BUILT_SOURCES)		\
-	acme-volume-alsa.c		\
-	acme-volume-alsa.h		\
+	acme-volume-dummy.c		\
+	acme-volume-dummy.h		\
 	acme-volume.c			\
 	acme-volume.h			\
 	eggaccelerators.c		\
@@ -99,8 +98,8 @@ SugarExt_1_0_gir_LIBS = libsugarext.la $
 SugarExt_1_0_gir_FILES = \
 	acme-volume.c \
 	acme-volume.h \
-	acme-volume-alsa.c \
-	acme-volume-alsa.h \
+	acme-volume-dummy.c \
+	acme-volume-dummy.h \
 	eggsmclient.c \
 	eggsmclient.h \
 	eggsmclient-xsmp.c \
--- origsrc/sugar-toolkit-gtk3-0.106.1/src/sugar3/acme-volume-dummy.c	1969-12-31 18:00:00.000000000 -0600
+++ src/sugar-toolkit-gtk3-0.106.1/src/sugar3/acme-volume-dummy.c	2015-12-30 02:58:06.580964400 -0600
@@ -0,0 +1,91 @@
+/* acme-volume-dummy.c
+
+   Copyright (C) 2002, 2003 Bastien Nocera
+
+   The Gnome Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   The Gnome Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the Gnome Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.
+
+   Author: Bastien Nocera <hadess@hadess.net>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "acme-volume-dummy.h"
+
+G_DEFINE_TYPE (AcmeVolumeDummy, acme_volume_dummy, ACME_TYPE_VOLUME)
+
+static void
+acme_volume_dummy_finalize (GObject *object)
+{
+	g_return_if_fail (object != NULL);
+	g_return_if_fail (ACME_IS_VOLUME_DUMMY (object));
+
+	G_OBJECT_CLASS (acme_volume_dummy_parent_class)->finalize (object);
+}
+
+static void
+acme_volume_dummy_set_mute (AcmeVolume *vol, gboolean val)
+{
+}
+
+static gboolean
+acme_volume_dummy_get_mute (AcmeVolume *vol)
+{
+	return FALSE;
+}
+
+static int
+acme_volume_dummy_get_volume (AcmeVolume *vol)
+{
+	return 0;
+}
+
+static void
+acme_volume_dummy_set_volume (AcmeVolume *vol, int val)
+{
+}
+
+/* minimum step size (in percent) required to actually affect volume */
+static int
+acme_volume_dummy_get_threshold (AcmeVolume *vol)
+{
+	return 1;
+}
+
+static void
+acme_volume_dummy_init (AcmeVolumeDummy *vol)
+{
+}
+
+AcmeVolumeDummy *
+acme_volume_dummy_new(gint type)
+{
+	AcmeVolumeDummy *obj = g_object_new(ACME_TYPE_VOLUME_DUMMY, NULL);
+	return obj;
+}
+
+static void
+acme_volume_dummy_class_init (AcmeVolumeDummyClass *klass)
+{
+	AcmeVolumeClass *volume_class = ACME_VOLUME_CLASS (klass);
+	G_OBJECT_CLASS (klass)->finalize = acme_volume_dummy_finalize;
+
+	volume_class->set_volume = acme_volume_dummy_set_volume;
+	volume_class->get_volume = acme_volume_dummy_get_volume;
+	volume_class->set_mute = acme_volume_dummy_set_mute;
+	volume_class->get_mute = acme_volume_dummy_get_mute;
+	volume_class->get_threshold = acme_volume_dummy_get_threshold;
+}
--- origsrc/sugar-toolkit-gtk3-0.106.1/src/sugar3/acme-volume-dummy.h	1969-12-31 18:00:00.000000000 -0600
+++ src/sugar-toolkit-gtk3-0.106.1/src/sugar3/acme-volume-dummy.h	2015-12-30 02:59:23.559367400 -0600
@@ -0,0 +1,45 @@
+/* acme-volume-dummy.h
+
+   Copyright (C) 2002, 2003 Bastien Nocera
+
+   The Gnome Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   The Gnome Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the Gnome Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.
+
+   Author: Bastien Nocera <hadess@hadess.net>
+ */
+
+#include <glib.h>
+#include <glib-object.h>
+#include "acme-volume.h"
+
+#define ACME_TYPE_VOLUME_DUMMY		(acme_volume_dummy_get_type ())
+#define ACME_VOLUME_DUMMY(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), ACME_TYPE_VOLUME_DUMMY, AcmeVolumeDummy))
+#define ACME_VOLUME_DUMMY_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), ACME_TYPE_VOLUME_DUMMY, AcmeVolumeDummyClass))
+#define ACME_IS_VOLUME_DUMMY(obj)	(G_TYPE_CHECK_INSTANCE_TYPE ((obj), ACME_TYPE_VOLUME_DUMMY))
+#define ACME_VOLUME_DUMMY_GET_CLASS(obj)	(G_TYPE_INSTANCE_GET_CLASS ((obj), ACME_TYPE_VOLUME_DUMMY, AcmeVolumeDummyClass))
+
+typedef struct AcmeVolumeDummy AcmeVolumeDummy;
+typedef struct AcmeVolumeDummyClass AcmeVolumeDummyClass;
+
+struct AcmeVolumeDummy {
+	AcmeVolume parent;
+};
+
+struct AcmeVolumeDummyClass {
+	AcmeVolumeClass parent;
+};
+
+GType acme_volume_dummy_get_type		(void);
+AcmeVolumeDummy *acme_volume_dummy_new		(gint);
